src/core/Browser.js

```diff
@@ -40,19 +40,21 @@ const touch = touchNative || pointer;
 
 // @property retina: Boolean
 // `true` for browsers on a high-resolution "retina" screen or on any screen when browser's display zoom is more than 100%.
-const retina = (window.devicePixelRatio || (window.screen.deviceXDPI / window.screen.logicalXDPI)) > 1;
+const retina = typeof window === 'undefined' || typeof window.devicePixelRatio === 'undefined' ? false : window.devicePixelRatio > 1;
 
 // @property mac: Boolean; `true` when the browser is running in a Mac platform
-const mac = navigator.platform.startsWith('Mac');
+const mac = typeof navigator === 'undefined' || typeof navigator.platform === 'undefined' ? false : navigator.platform.startsWith('Mac');
 
 // @property mac: Boolean; `true` when the browser is running in a Linux platform
-const linux = navigator.platform.startsWith('Linux');
+const linux = typeof navigator === 'undefined' || typeof navigator.platform === 'undefined' ? false : navigator.platform.startsWith('Linux');

```

Comment:
Not sure if this can be handled by the SSR framworks, but this could be a bit more terse using nullish coalescing. 
```suggestion
const linux = typeof navigator?.plaform === 'undefined' ? false : navigator.platform.startsWith('Linux');

```

Reply:
Node can't handle that:
![grafik](https://github.com/Leaflet/Leaflet/assets/19800037/5f698930-6b30-40eb-97e8-d38947d659b6)

Reply:
Figured as much, let's keep it like this then